numpy/typing/tests/data/reveal/ma.pyi

```diff
@@ -1,14 +1,21 @@
-from typing import Any, Literal, TypeAlias, TypeVar, assert_type, NoReturn
+from typing import Any, Literal, TypeAlias, TypeVar, assert_type, NoReturn, Generic
 
 import numpy as np
 from numpy import dtype, generic
 from numpy._typing import NDArray, _AnyShape
 
 _ScalarT = TypeVar("_ScalarT", bound=generic)
+_ScalarT_co = TypeVar("_ScalarT_co", bound=generic, covariant=True)
 MaskedArray: TypeAlias = np.ma.MaskedArray[_AnyShape, dtype[_ScalarT]]
 _Array1D: TypeAlias = np.ndarray[tuple[int], np.dtype[_ScalarT]]
 
-class MaskedArraySubclass(MaskedArray[np.complex128]): ...
+class MaskedArraySubclass(MaskedArray[_ScalarT_co]): ...
+
+class IntoMaskedArraySubClass(Generic[_ScalarT_co]):
+    def __array__(self) -> MaskedArraySubclass[_ScalarT_co]:
+        ...
```

Comment:
```suggestion
    def __array__(self) -> MaskedArraySubclass[_ScalarT_co]: ...
```

(the `Into*` in the name makes me think that you know rust, is that correct :P ?)